torchx/lib/torchx/backend.ex

```diff
@@ -125,11 +125,56 @@ defmodule Torchx.Backend do
 
   ## Transfer
 
-  # TODO: Handle backend_options
   @impl true
-  def to_batched_list(%T{shape: shape} = out, %T{} = t, _backend_options) do
-    Torchx.split(from_nx(t), elem(shape, 0))
-    |> Enum.map(&to_nx(&1, out))
+  def to_batched_list(%T{shape: shape} = out, %T{} = t, opts) do
+    leftover = opts[:leftover]
+
+    batch_size = elem(shape, 0)
+    t_axis_0 = elem(t.shape, 0)
+
+    remainder = rem(t_axis_0, batch_size)
+    num_full_batches = div(t_axis_0, batch_size)
+
+    num_batches =
+      if leftover == :repeat and remainder != 0 do
+        num_full_batches + 1
+      else
+        num_full_batches
+      end
+
+    to_batch =
+      cond do
+        remainder != 0 and leftover == :discard ->
+          t
+
+        remainder != 0 and leftover == :repeat ->
+          concat_shape =
+            t.shape |> Tuple.delete_at(0) |> Tuple.insert_at(0, num_batches * batch_size)
+
+          slice_length = t_axis_0 - remainder
+          slice_shape = t.shape |> Tuple.delete_at(0) |> Tuple.insert_at(0, slice_length)
+
+          slice = slice(%{t | shape: slice_shape}, t, [0], [slice_length], [1])
+
+          concatenate(%{t | shape: concat_shape}, [t, slice], 0)
```

Comment:
My suggestion: avoid calling the backend itself because we need to do all of these shapes manipulation. Instead, call the Torchx.xyz function directly. If necessary, encapsulate the logic behind `Torchx.Backend.slice` into a private `torchx_slice` so you can reuse it here. This will hopefully make things easier to debug too. :)